src/ui/pages/SidePage/components/IncomingRequest/IncomingRequest.tsx

```diff
@@ -12,24 +12,35 @@ import {
   IncomingRequestProps,
   IncomingRequestType,
 } from "../../../../../store/reducers/stateCache/stateCache.types";
+import { getConnectedWallet } from "../../../../../store/reducers/walletConnectionsCache";
 
 const IncomingRequest = ({ open, setOpenPage }: SidePageContentProps) => {
   const pageId = "incoming-request";
   const dispatch = useAppDispatch();
   const queueIncomingRequest = useAppSelector(getQueueIncomingRequest);
+  const connectedWallet = useAppSelector(getConnectedWallet);
   const incomingRequest = useMemo(() => {
-    return !queueIncomingRequest.isProcessing
-      ? { id: "" }
-      : queueIncomingRequest.queues.length > 0
-        ? queueIncomingRequest.queues[0]
-        : { id: "" };
+    if (
+      !queueIncomingRequest.isProcessing ||
+      !queueIncomingRequest.queues.length
+    ) {
+      return { id: "" };
+    } else {
+      return queueIncomingRequest.queues[0];
+    }
   }, [queueIncomingRequest]);
   const [initiateAnimation, setInitiateAnimation] = useState(false);
   const [requestData, setRequestData] = useState<IncomingRequestProps>();
   const ANIMATION_DELAY = 4000;
   const [blur, setBlur] = useState(false);
 
   useEffect(() => {
+    if (
+      incomingRequest.type === IncomingRequestType.PEER_CONNECT_SIGN &&
+      !connectedWallet
```

Comment:
Can we explicitly check the ID of the connected wallet here? Or is that info not available

Just in case we connect to 2 dApps and have a queue mixed with sign requests

Reply:
I already handled the incoming requests when the connection is no longer connected, so I think we don't need to check the ID here.

Reply:
But what if:
- you have a queue of items for a dApp
- you disconnect
- you connect to a _different_ dApp and stay connected

This won't catch that.